src/hooks/core/chat/useMoveChatTypeHook.ts

```diff
@@ -94,8 +83,9 @@ const useMoveChatTypeHook = () => {
     }
 
     try {
+      const members = channel?.better_channel_member || channel?.members || [];
       await Promise.all(
-        (channel?.members || []).map(async (member) => {
+        members?.map(async (member) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The members are now fetched from `channel.better_channel_member` if it exists, otherwise from `channel.members`. Ensure that `channel.better_channel_member` always contains valid data when it exists. If it can be undefined or null, add checks to handle these cases.



<!-- This is an auto-generated comment by OSS CodeRabbit -->